File: 136-single-number/single-number.java
class Solution {
    public int singleNumber(int[] nums) {
        int result = 0;
        // for each loop to compute XOR for all elements 
        // at last we get single element bcoz other appears twice so 2^2 = 0 & num ^ 0 = num
        for(int num: nums){
            result ^= num;
        }

        return result;
    }
}

File: 136-single-number/single-number.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        if len(nums) == 1:
            return nums[0]
        
        freq = {}
        for ele in nums:
            freq[ele] = freq.get(ele, 0) + 1

        for key, value in freq.items():
            if value == 1:
                return key

        return